scripts/brp-strip

```diff
@@ -7,11 +8,37 @@ fi
 STRIP=${1:-strip}
 NCPUS=${RPM_BUILD_NCPUS:-1}
 
+# 32 was chosen as a compromise between reducing the overhead of starting new
+# processes and distributing the work load evenly over as much processors as
+# possible
+MAX_ARGS=32
+
 case `uname -a` in
 Darwin*) exit 0 ;;
 *) ;;
 esac
 
 # Strip ELF binaries
-find "$RPM_BUILD_ROOT" -type f \! -regex "${RPM_BUILD_ROOT}/*usr/lib/debug.*" \! -name "*.go" -print0 | \
-    xargs -0 -r -P$NCPUS -n32 sh -c "file \"\$@\" | sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p' | grep -v 'no machine' | xargs -I\{\} $STRIP -g \{\}" ARG0
+# Below is the explanation of commands in the order of their appearance
+# Ignore /usr/lib/debug entries
+# Ignore all go(guile objects & golang) files
+# Consider files with only single link
+# Run the file command to find relevant non-stripped binaries, with bundle size of 32
+# Ignore all 'no machine' files
+# Only operate on non-stripped binaries
+find "$RPM_BUILD_ROOT" -type f \
+  ! -regex "${RPM_BUILD_ROOT}/*usr/lib/debug.*" \
+  ! -name "*.go" -links 1 -print0 | \
+  xargs -0 -r -P${NCPUS} -n${MAX_ARGS} sh -c "file \"\$@\" | \
+  sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p' | \
+  grep -v 'no machine' | \
+  xargs -I\{\} $STRIP -g \{\}" ARG0
+
+# Same as above but for files with more than 1 link
+for f in $(find "$RPM_BUILD_ROOT" -type f \
+          ! -regex "${RPM_BUILD_ROOT}/*usr/lib/debug.*" \
+          ! -name "*.go" -links +1 -exec file {} \; | \
+          sed -n -e 's/^\(.*\):[  ]*ELF.*, not stripped.*/\1/p' | \
+          grep -v 'no machine'); do
+  $STRIP -g "$f" || :
```

Comment:
The two commands now differ fundamentally, the latter being subject to issues with whitespace in filenames etc and any change to the logic now needs to be done twice, to a different command, one of which doesn't trigger at all for most packages. Which is a recipe for problems.
 
Put the parallel-aware command into a function which takes the limits as arguments, and never mind that non-parallel run will be slightly more complicated than it strictly needs to be. The calling would then look basically like this:

```
strip_files 1 ${NCPUS}
strip_files +1 1
```

Reply:
Nice idea. Thanks :)

Reply:
@pmatilai I have made the suggested changes, please review again.